File: src/main/java/me/sd_master92/customvoting/commands/SettingsCommand.java
package me.sd_master92.customvoting.commands;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.constants.enumerations.SoundType;
import me.sd_master92.customvoting.gui.VoteSettings;
import me.sd_master92.plugin.command.SimpleCommand;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;

public class SettingsCommand extends SimpleCommand
{
    private final Main plugin;

    public SettingsCommand(Main plugin)
    {
        super(plugin, "votesettings");
        withPlayer();
        this.plugin = plugin;
    }

    @Override
    public void onCommand(CommandSender commandSender, String[] strings)
    {

    }

    @Override
    public void onCommand(Player player, String[] strings)
    {
        Inventory settings = new VoteSettings().getInventory();
        SoundType.OPEN.play(plugin, player);
        player.openInventory(settings);
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/commands/SettingsCommand.kt
package me.sd_master92.customvoting.commands

import me.sd_master92.core.command.SimpleCommand
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.Message
import me.sd_master92.customvoting.constants.enumerations.SoundType
import me.sd_master92.customvoting.gui.pages.settings.VoteSettingsPage
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class SettingsCommand(private val plugin: CV) : SimpleCommand(plugin, "votesettings")
{
    override fun onCommand(sender: CommandSender, args: Array<String>)
    {
    }

    override fun onCommand(player: Player, args: Array<String>)
    {
        SoundType.OPEN.play(plugin, player)
        VoteSettingsPage(plugin, null).open(player)
    }

    init
    {
        withPlayer(Message.MUST_BE_PLAYER.getMessage(plugin))
        withNoPermMessage(Message.NO_PERMISSION.getMessage(plugin))
    }
}